src/Command.js

```diff
@@ -147,12 +148,26 @@ export default class Command {
       exitHandler.writeLogs();
     }
 
-    if (callback) {
-      callback(err, code);
+    const finish = function() {
+      if (callback) {
+        callback(err, code);
+      }
+
+      if (process.env.NODE_ENV !== "test") {
+        process.exit(code);
+      }
     }
 
-    if (process.env.NODE_ENV !== "test") {
-      process.exit(code);
+    const childProcessCount = ChildProcessUtilities.getChildProcessCount();
+    if (childProcessCount > 0) {
+      logger.info(
+        `Waiting for ${childProcessCount} child ` +
+        `process${childProcessCount === 1 ? "" : "es"} to exit. ` +
+        "CTRL-C to exit immediately."
+      );
+      ChildProcessUtilities.onAllExited(finish);
+    } else {
+      finish();
     }
```

Comment:
Instead of writing checks like this, it'd be easier to have an API:

```js
ChildProcessUtilities.poorlyNamed__maybeWaitForEverythingToExit(finish);
```

where `poorlyNamed__maybeWaitForEverythingToExit` did all of the above for you.